### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by jvanderleeuw**
made changes from pull request for updateHOW_TO branch

### CODE REVIEW

File: algorithm_rgb.py [hunk 1]
```diff
@@ -48,15 +48,122 @@
...
+    """
+    Returns the percentage of an image that is green, which can be used
+    to identify plant cover
+    """
+    red, green, blue = calculate(pxarray)
+
+    return green / (red + green + blue)
+
+
 def calculate(pxarray: np.ndarray):
```
Comment (Chris-Schnaufer) [inline algorithm_rgb.py]: You are using this incorrectly. Please look at the HOWTO

File: pylint.rc [hunk 2]
```diff
@@ -0,0 +1,91 @@
+# Version of this pylint.rc file
```
Comment (Chris-Schnaufer) [inline pylint.rc]: This file does not belong here. It's in the Organization repo. I can see how the README for Organization repo is confusing, it will be corrected.

**Review by Chris-Schnaufer — COMMENTED** [review]
Still have extra files in this PR

File: pylint.rc (hunk 2 above)
Comment (Chris-Schnaufer) [reply pylint.rc]: I updated the documentation and made a PR

File: algorithm_rgb.py (hunk 1 above)
Comment (jvanderleeuw) [reply algorithm_rgb.py]: Didn't realize I still had the "--> float" in that function. I deleted it

Comment (jvanderleeuw) [reply algorithm_rgb.py]: I was going off of lines 41-44 in the HOW_TO

Comment (Chris-Schnaufer) [reply algorithm_rgb.py]: You can replace the "-> float" with "-> tuple".

Comment (Chris-Schnaufer) [reply algorithm_rgb.py]: That's what you want the algorithm to return as the results of the calculations: the averages of the red, green, and blue channels?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,31 @@
+# automatically generated: 2020-05-20T08:40:04.406489
+FROM agpipeline/rgb-plot-base-image:latest
+LABEL maintainer="Unknown <Unknown@gmail.com>"
```
Comment (Chris-Schnaufer) [inline Dockerfile]: This should be your info

File: README.md [hunk 4]
```diff
@@ -2,7 +2,8 @@
 Description <!-- Replace this entire line with a short description of your algorithm -->
```
Comment (Chris-Schnaufer) [inline README.md]: This still needs to be filled out per previous comment regarding that

File: algorithm_rgb.py (hunk 1 above)
Comment (jvanderleeuw) [reply algorithm_rgb.py]: I asked David but he hasn't responded yet. Based on what the function looks like in the google doc, I would assume that it is the entire amount of each type of pixel rather than the averages. I can change it to that for now and revert if he says otherwise

Comment (Chris-Schnaufer) [reply algorithm_rgb.py]: You need this function to return the results of the calculations, not the averages of each color channel.

File: algorithm_rgb.py [hunk 5]
```diff
@@ -48,15 +48,122 @@
```
Comment (Chris-Schnaufer) [inline algorithm_rgb.py:43]: You still need to assign valid strings to the global variables above this line (lint 49)

File: algorithm_rgb.py (hunk 1 above)
Comment (jvanderleeuw) [reply algorithm_rgb.py]: As in all of the calculations would be returned by the one function call? I'll do that

**Comment by jvanderleeuw** [comment]
Made edits to files and added in descriptions

**Comment by jvanderleeuw** [comment]
@dlebauer made changes to README

File: README.md [hunk 6]
```diff
@@ -1,17 +1,51 @@
...
+
+## Overview
+The Greenness Transformer is meant to provide several calculations on image rgb data in order to get an idea of plant growth and
+health. A more in-depth explanation can be found at [this link](https://docs.google.com/document/d/1cAm5w1Bs6dB1SHgf-HVmwwbebLhbZMUvTkXLtas_-xI/edit)
+
+## Algorithm Description
+This transformer provides calculations using the red, green, and blue values for pixels in an image ranging from calculating the
+percentage of an image which is green to normalized difference calculations and more
 
 ## References
```
Comment (Chris-Schnaufer) [inline README.md:17]: Can you adjust the headings below to provide clarity as to what goes with what. Having everything at the same level makes it difficult to read. Any approach that clarifies the relationships between the text is fine.

File: algorithm_rgb.py [hunk 7]
```diff
@@ -1,62 +1,184 @@
...
+    return round(2 * green - (red + blue), 2)
+
+
+def green_leaf_index(pxarray: np.ndarray) -> float:
+    """
+    Calculates the green leaf index of an image
+    """
+    red, green, blue = get_red_green_blue_averages(pxarray)
+
+    return round((2 * green - red - blue) / (2 * green + red + blue), 2)
```
Comment (Chris-Schnaufer) [inline algorithm_rgb.py:64]: Should check for zero denominator and return zero so that an exception isn't thrown. Here and below

File: algorithm_rgb.py [hunk 8]
```diff
@@ -1,62 +1,184 @@
...
-# For each variable name in VARIABLE_NAMES add the unit of measurement the value represents.
-# !! Replace the content of this string with your variables' unit
-VARIABLE_UNITS = 'pixels'
+VARIABLE_UNITS = '[-510:510], [-1:1], [-255:255], [-127:129], [-255:255], [-255:332], ' \
+                 '[-1000:1000], [-1000:1000], [-255:255], [-255:255], [0:100]'
 
 # Variable labels matching the order of VARIABLE_NAMES, also comma-separated.
-# This is an optional definition and can be left empty.
-VARIABLE_LABELS = ''
+VARIABLE_LABELS = 'excess_greenness_index, green_leaf_index, cive, normalized_difference_index(pxarray), ' \
```
Comment (Chris-Schnaufer) [inline algorithm_rgb.py:35]: why is the variable named 'normalized_difference_index(pxarray)' when it's a float? Shouldn't `(pxarray)` be removed?

File: testing.py [hunk 9]
```diff
@@ -25,9 +25,11 @@ def _get_variables_header_fields() -> str:
     if labels_len != len(variables):
         sys.stderr.write("The number of defined labels doesn't match the number of defined variables")
         sys.stderr.write("  continuing processing")
+        sys.stderr.write("\n")
```
Comment (Chris-Schnaufer) [inline testing.py:28]: Why this change? Should it be a general change in the template repo?

**Review by Chris-Schnaufer — COMMENTED** [review]
Quite the change. Looking good & almost there

File: testing.py (hunk 9 above)
Comment (jvanderleeuw) [reply testing.py:28]: I added that change because it is hard to read when neither labels_len or units_len is equal to len(variables) since both print statements are written on the same line. I changed it so that the \n is at the end of each line so that there does not need to be a separate sys.stderr.write for each block

File: algorithm_rgb.py (hunk 8 above)
Comment (jvanderleeuw) [reply algorithm_rgb.py:35]: just removed; need to push still

File: algorithm_rgb.py (hunk 7 above)
Comment (jvanderleeuw) [reply algorithm_rgb.py:64]: updated and tested with all black and all white GIMP images
